source/unicodeScriptData.py

```diff
@@ -804,33 +850,63 @@
 	( 0X1f100 , 0X1f10c , "Common" ), 
 	( 0X1f110 , 0X1f12e , "Common" ), 
 	( 0X1f130 , 0X1f16b , "Common" ), 
-	( 0X1f170 , 0X1f19a , "Common" ), 
+	( 0X1f170 , 0X1f1ac , "Common" ), 
 	( 0X1f1e6 , 0X1f1ff , "Common" ), 
 	( 0X1f200 , 0X1f200 , "Hiragana" ), 
 	( 0X1f201 , 0X1f202 , "Common" ), 
-	( 0X1f210 , 0X1f23a , "Common" ), 
+	( 0X1f210 , 0X1f23b , "Common" ), 
 	( 0X1f240 , 0X1f248 , "Common" ), 
 	( 0X1f250 , 0X1f251 , "Common" ), 
-	( 0X1f300 , 0X1f579 , "Common" ), 
-	( 0X1f57b , 0X1f5a3 , "Common" ), 
-	( 0X1f5a5 , 0X1f6d0 , "Common" ), 
+	( 0X1f260 , 0X1f265 , "Common" ), 
+	( 0X1f300 , 0X1f6d4 , "Common" ), 
 	( 0X1f6e0 , 0X1f6ec , "Common" ), 
-	( 0X1f6f0 , 0X1f6f3 , "Common" ), 
+	( 0X1f6f0 , 0X1f6f8 , "Common" ), 
 	( 0X1f700 , 0X1f773 , "Common" ), 
 	( 0X1f780 , 0X1f7d4 , "Common" ), 
 	( 0X1f800 , 0X1f80b , "Common" ), 
 	( 0X1f810 , 0X1f847 , "Common" ), 
 	( 0X1f850 , 0X1f859 , "Common" ), 
 	( 0X1f860 , 0X1f887 , "Common" ), 
 	( 0X1f890 , 0X1f8ad , "Common" ), 
-	( 0X1f910 , 0X1f918 , "Common" ), 
-	( 0X1f980 , 0X1f984 , "Common" ), 
+	( 0X1f900 , 0X1f90b , "Common" ), 
+	( 0X1f910 , 0X1f93e , "Common" ), 
+	( 0X1f940 , 0X1f94c , "Common" ), 
+	( 0X1f950 , 0X1f96b , "Common" ), 
+	( 0X1f980 , 0X1f997 , "Common" ), 
 	( 0X1f9c0 , 0X1f9c0 , "Common" ), 
+	( 0X1f9d0 , 0X1f9e6 , "Common" ), 
 	( 0X20000 , 0X2a6d6 , "Han" ), 
 	( 0X2a700 , 0X2b734 , "Han" ), 
 	( 0X2b740 , 0X2b81d , "Han" ), 
 	( 0X2b820 , 0X2cea1 , "Han" ), 
+	( 0X2ceb0 , 0X2ebe0 , "Han" ), 
 	( 0X2f800 , 0X2fa1d , "Han" ), 
 	( 0Xe0001 , 0Xe0001 , "Common" ), 
 	( 0Xe0020 , 0Xe007f , "Common" ), 
 ]
+
+def getScriptCode(chr):
+	"""performs a binary search in scripCodes for unicode ranges
+	@param chr: character for which a script should be found
+	@type chr: string
+	@return: script code
+	@rtype: int"""
+	mStart = 0
+	mEnd = len(scriptRanges)-1
+	characterUnicodeCode = ord(chr)
+	# Number should respect preferred language setting
+	# FullWidthNumber is in Common category, however, it indicates Japanese language context
+	if DIGIT_ZERO  <= characterUnicodeCode <= DIGIT_NINE:
```

Comment:
Rather than having a special condition for "Number" and "FullWidthNumber", I think its cleaner to make these entries in the `scriptRanges`. Add these two entries explicitly from `unicodeScriptPrep`. Actually, this will require ensuring that there is no overlap with other entries which may be tricky to do. So maybe not.

Reply:
What do you think of this suggestion? It would be interesting to test the performance of the `getScriptFunction` without the two special cases, and manually put those ranges for numbers in the `scriptRanges` list.